nemo/collections/asr/parts/preprocessing/perturb.py

```diff
@@ -558,11 +565,11 @@ def perturb_with_foreground_noise(self, data, noise, data_rms=None, max_noise_du
             noise_gain_db = data_rms - noise.rms_db - snr_db
         noise_gain_db = min(noise_gain_db, self._max_gain_db)
 
-        n_additions = self._rng.randint(1, max_additions)
+        n_additions = random.randint(1, max_additions)
 
         for i in range(n_additions):
-            noise_dur = self._rng.uniform(0.0, max_noise_dur)
-            start_time = self._rng.uniform(0.0, noise.duration)
+            noise_dur = random.uniform(0.0, max_noise_dur)
```

Comment:
do I understand correctly that the random state is now global? is it more reproducible with global state rather than local?

Reply:
Yes the random state is now global (if no specific random seed is provided for each perturb), and from the results it seems that now it can fully reproduce the same results with the same random seed.

Could @fayejf please provide more explanations/details?

Reply:
Well, maybe it now reproduces, but it's then more brittle in how it interacts with other code potentially changing the global random state or resetting it...

Reply:
Thanks for your question and suggestion. In which case do you think it will potentially change the global random state or reset it? Could you elaborate more please?

Reply:
I meant that if some other aug in the pipeline also resets the seed, then it might break the reproducibility of this aug because they both affect the same global state.

In other places maybe I saw the other way around and torch rng generators are introduced as explicit arguments to fully control the randomness (e.g. this is being discussed in torchvision)

But it was just a question :) I haven't read very deeply the NeMo pipeline, and I don't have good, well-based arguments etc...